lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -2908,13 +2890,30 @@ SwiftASTContext::CreateInstance(const SymbolContext &sc,
     ArchSpec preferred_arch;
     llvm::Triple preferred_triple;
     if (is_repl) {
+      LOG_PRINTF(GetLog(LLDBLog::Types), "REPL: prefer target triple.");
+      preferred_arch = target_arch;
+      preferred_triple = target_triple;
+    } else if (!sdk_path_override && !sdk) {
+      LOG_PRINTF(GetLog(LLDBLog::Types),
+                 "No Swift debug info: prefer target triple.");
+      if (!target_arch.IsCompatibleMatch(module_arch))
+        HEALTH_LOG_PRINTF(
+            "SwiftASTContext requested for a non-Swift translation unit. Using "
+            "target triple \"%s\", which is not compatible with this "
+            "translation unit's triple \"%s\". Expressions may behave "
+            "unexpectedly because of this.",
+            target_triple.str().c_str(), module_triple.str().c_str());
```

Comment:
Is this too noisy to report as a Diagnostic? This seems pretty valuable and something we could deduplicate at the `Module` level perhaps?

Reply:
In my first draft, I had this as Debugger::ReportWarning. But the problem I see is that it could be really hard to associate the warning with the action that triggered it. You get the warning when the first Swift type system in that SymbolContext is initialized, but that might be triggered by, eg., your IDE's variable view.